src/components/shared/GitHubAuth/AuthorizedUserProfile.tsx

```diff
@@ -0,0 +1,65 @@
+import { LogOutIcon } from "lucide-react";
+import { useCallback, useSyncExternalStore } from "react";
+
+import TooltipButton from "../Buttons/TooltipButton";
+import { useAuthLocalStorage } from "./useAuthLocalStorage";
+
+export function AuthorizedUserProfile() {
+  const localTokenStorage = useAuthLocalStorage();
+
+  /**
+   * This is a workaround to avoid the useSyncExternalStore from updating infinitely.
+   *
+   * `getToken` is used to trigger the useSyncExternalStore to update the profile.
+   * Profile is parsed from the JSON in the local storage, resulting in a new object every time,
+   *   which triggers the useSyncExternalStore to update infinitely.
+   */
+  const token = useSyncExternalStore(
+    localTokenStorage.subscribe,
+    localTokenStorage.getToken,
+  );
+  const profile = localTokenStorage.getJWT();
+
+  const handleLogout = useCallback(() => {
+    localTokenStorage.clear();
+  }, [localTokenStorage]);
+
+  if (!token) {
+    return null;
+  }
+
+  if (!profile) {
+    return null;
+  }
```

Comment:
```suggestion
  if (!token || !profile) {
    return null;
  }
```

There's an argument to be made to keep them separate, or to combine them. I don't mind either way but putting this here to see what the team prefers